tests/run-make/wasm-abi/rmake.rs

```diff
@@ -0,0 +1,43 @@
+extern crate run_make_support;
+
+use run_make_support::{out_dir, rustc};
+use std::path::Path;
+use std::process::Command;
+
+fn main() {
+    if std::env::var("TARGET").unwrap() != "wasm32-wasip1" {
+        return;
+    }
+
+    rustc().arg("foo.rs").arg("--target=wasm32-wasip1").run();
+    let file = out_dir().join("foo.wasm");
+
+    let has_wasmtime = match Command::new("wasmtime").arg("--version").output() {
+        Ok(s) => s.status.success(),
+        _ => false,
+    };
+    if !has_wasmtime {
+        println!("skipping test, wasmtime isn't available");
```

Comment:
This will be entirely hidden when running `./x.py test`. It will not even be marked as ignored test.

Reply:
> True! I don't know of a great way to manage that, but I suppose an alternative would be to require a `RUNNER` style env var and to consult that?

I'm a complete idiot and forgor that when implementing support for rmake.rs, compiletest directives **are** supported in rmake.rs! So this could be actually implemented in compiletest as a `//@ needs-wasmtime` header